src/test/java/io/confluent/examples/streams/SumLambdaIntegrationTest.java

```diff
@@ -89,33 +79,29 @@ public void shouldSumEvenNumbers() throws Exception {
         .reduce((v1, v2) -> v1 + v2);
     sumOfOddNumbers.toStream().to(outputTopic);
 
-    final KafkaStreams streams = new KafkaStreams(builder.build(), streamsConfiguration);
-    streams.start();
+    final TopologyTestDriver topologyTestDriver = new TopologyTestDriver(builder.build(), streamsConfiguration);
 
     //
     // Step 2: Produce some input data to the input topic.
     //
-    final Properties producerConfig = new Properties();
-    producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
-    producerConfig.put(ProducerConfig.ACKS_CONFIG, "all");
-    producerConfig.put(ProducerConfig.RETRIES_CONFIG, 0);
-    producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class);
-    producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class);
-    IntegrationTestUtils.produceValuesSynchronously(inputTopic, inputValues, producerConfig);
+    IntegrationTestUtils.produceKeyValuesSynchronously(
+      inputTopic,
+      inputValues.stream().map(i -> new KeyValue<Void, Integer>(null, i)).collect(Collectors.toList()),
+      topologyTestDriver,
+      new IntegrationTestUtils.NothingSerde<>(true),
+      new IntegerSerializer()
+    );
 
     //
     // Step 3: Verify the application's output data.
     //
-    final Properties consumerConfig = new Properties();
-    consumerConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
-    consumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, "sum-lambda-integration-test-standard-consumer");
-    consumerConfig.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
-    consumerConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, IntegerDeserializer.class);
-    consumerConfig.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, IntegerDeserializer.class);
-    final List<String> actualValues = IntegrationTestUtils.waitUntilMinValuesRecordsReceived(consumerConfig,
-        outputTopic, expectedValues.size());
-    streams.close();
-    assertThat(actualValues).isEqualTo(expectedValues);
+    final Collection<Integer> actualValues = IntegrationTestUtils.drainTableOutput(
+      outputTopic,
+      topologyTestDriver,
+      new IntegerDeserializer(),
+      new IntegerDeserializer()
+    ).values();
+    assertThat(actualValues).containsExactlyElementsOf(expectedValues);
```

Comment:
Why no `isEquals`?

Reply:
I'll try it again. I think I tried it, but it might have been `is()`. It failed because the expected/actual collections had different concrete types.

Reply:
Weird. It worked with no problem. Thanks for the catch.